### ISSUE TITLE
Frame #1 for rxresp was of type WINDOW_UPDATE (8) instead of HEADERS (1)

### ISSUE THREAD

**Issue by Dridi** [varnishcache/varnish-cache#3442]
Sometimes WINDOW_UPDATE comes first, sometimes HEADERS:

```vcl
varnishtest "h2 rxresp winup race"

barrier b1 sock 2

varnish v1 -cliok "param.set feature +http2"
varnish v1 -vcl {
        import vtc;

        backend be none;

        sub vcl_recv {
                return (synth(200));
        }

        sub vcl_synth {
                vtc.barrier_sync("${b1_sock}");
        }
} -start

client c1 {
        stream 1 {
                txreq -method RACE -hdr content-length 1024 -nostrend
                barrier b1 sync
                delay 0.1
                txdata -datalen 1024
                #rxwin
                rxresp
                expect resp.status == 200
        } -run
} -run
```

I haven't checked yet how we're supposed to behave, spotted while working on #3441.

**Comment by Dridi** [comment]
After some research with @daghf's help I realize that this is caused by the default value of our parameters and the huge window we offer to request bodies by default.

The problem is solved for #3441 with https://github.com/varnishcache/varnish-cache/pull/3441/commits/1f57f6fd00e434c252671cf900c86514747ef081.

**Comment by Dridi** [comment]
I think we should solve this by having less aggressive defaults in test cases.

**Comment by Dridi** [comment]
As agreed during today's bugwash, I made varnishtest use more generous defaults to give some slack h2 test cases that aren't meant to actively cover window updates in c9fc384.

**Comment by Dridi** [comment]
I think the way forward should be to have an automatic window update handling, on by default, that we can disable when we are explicitly testing window updates.

**Comment by Dridi** [comment]
And with #3661 merged this is now fixed in 7285491019712223ca1c33b873d0d21362a35107.
